plugin/k8s_external/external.go

```diff
@@ -93,6 +102,10 @@ func (e *External) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 	m.Authoritative = true
 
 	if len(svc) == 0 {
+		if e.Fall.Through(state.Name()) {
```

Comment:
The behavior for `fallthough` we try to adhere to globally in CoreDNS is to only fall through when the domain queried does not exist, and not to fall through if the domain exists but hold no records of the queried type ...

```suggestion
		if e.Fall.Through(state.Name()) && rcode == dns.RcodeNameError {
```

Reply:
Thank you for your feedback! It appears that this proposed change prioritizes the accuracy and reliability of DNS queries.